src/source.rs

```diff
@@ -138,23 +160,36 @@ impl<I: AsRef<str>> From<I> for Source<I> {
             text: input,
             lines,
             len: char_offset,
-            byte_len: byte_offset
+            byte_len: byte_offset,
         }
     }
 }
 
 impl<I: AsRef<str>> Source<I> {
     /// Get the length of the total number of characters in the source.
-    pub fn len(&self) -> usize { self.len }
+    pub fn len(&self) -> usize {
+        self.len
+    }
+
+    /// Returns `true` if this source contains no characters.
+    pub fn is_empty(&self) -> bool {
+        self.len() == 0
+    }
```

Comment:
Clippy sais that if you implement `len` you must also implement `is_empty`. This has no serious impact.